### PR TITLE
ref(open-pr-comment): pass repo_id when finding codemappings

### PR DESCRIPTION
**Pull request by cathteng**
We are having some cases where our Snuba query to fetch unhandled issues is too big ([example](https://sentry.sentry.io/issues/4929543284/)). This results in the error, and a comment is not made on an open PR.

The exact reason is because we are passing too many recent, unresolved issue ids into the query. If we backtrack from this reason, another reason is that we have too many projects in which we are fetching issues from.

If we backtrack even further, this could be happening because too many codemappings match when we are trying to go from the filename in Github to how the filename is represented in Sentry. We get projects from codemappings because each codemapping is associated with a project. A single codemapping is also mapped to a single repo -- and we have a repository from the webhook that kicks off an open PR comment.

tl;dr We can pass in the repository id into our search for matching codemappings. This makes the list of projects that we obtain from the matching codemappings more accurate, and should reduce the number of recent unresolved issues that we pass into the Snuba query.

### CODE REVIEW

File: tests/sentry/tasks/integrations/github/test_open_pr_comment.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def test_get_projects_and_filenames_from_source_file(self):
         ]
 
         project_list, sentry_filenames = get_projects_and_filenames_from_source_file(
-            self.organization.id, filename
+            self.organization.id, self.gh_repo.id, filename
```
Comment (nhsiehgit) [inline tests/sentry/tasks/integrations/github/test_open_pr_comment.py:331]: Can we have a separate more explicit test for filtering by repo id?

Comment (cathteng) [reply tests/sentry/tasks/integrations/github/test_open_pr_comment.py:331]: sure

Comment (cathteng) [reply tests/sentry/tasks/integrations/github/test_open_pr_comment.py:331]: done!

**Review by nhsiehgit — APPROVED** [review]
